### PR TITLE
Count modal in TS

### PR DESCRIPTION
**Pull request by JulianWielga**
fixed few things in cypress tests. now there are separated snapshots pew pixel ratio. added custom commands to prepare/cleanup test processes.
added test for counts modal.
refactor of counts modal - moved to ts and to functional component

### CODE REVIEW

File: ui/client/components/modals/CalculateCountsDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+const dateFormat = "YYYY-MM-DD"
+const timeFormat = "HH:mm:ss"
+
+interface Range {
+  name: string,
+  from: () => Moment,
+  to: () => Moment,
+}
+
+const predefinedRanges: Range[] = [
```
Comment (lciolecki) [inline ui/client/components/modals/CalculateCountsDialog.tsx:25]: Missing translations at name param.

File: ui/client/components/modals/CalculateCountsDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+  const setRange = useCallback((value: [MomentInput, MomentInput]) => {
+    const [from, to] = value.map(v => moment(v).toDate())
+    setState(current => ({...current, from, to}))
+  }, [])
+
+  return (
+    <GenericModalDialog init={init} confirm={confirm} type={Dialogs.types.calculateCounts}>
+      <Picker
+        label="Process counts from"
```
Comment (lciolecki) [inline ui/client/components/modals/CalculateCountsDialog.tsx:133]: Missing translation.

File: ui/client/cypress.json [hunk 3]
```diff
@@ -7,8 +7,13 @@
   "env": {
     "cypress-plugin-snapshots": {
       "imageConfig": {
-        "threshold": 0.01
-      }
+        "resizeDevicePixelRatio": false,
+        "threshold": 0.001
```
Comment (lciolecki) [inline ui/client/cypress.json:11]: Can you add `,` on end?

File: ui/client/cypress.json [hunk 4]
```diff
@@ -7,8 +7,13 @@
...
+      },
+      "screenshotConfig": {
+        "scale": false
```
Comment (lciolecki) [inline ui/client/cypress.json:14]: The same as above.

File: ui/client/cypress.json [hunk 5]
```diff
@@ -7,8 +7,13 @@
...
+      },
+      "backgroundBlend": "overlay"
```
Comment (lciolecki) [inline ui/client/cypress.json:16]: The same as above.

**Review by lciolecki — CHANGES_REQUESTED** [review]
What is it `/ui/client/cypress/integration/__image_snapshots__/` ? Why files in this path (.png) are in repository?

File: ui/client/cypress.json (hunk 4 above)
Comment (JulianWielga) [reply ui/client/cypress.json:14]: its json

File: ui/client/cypress.json (hunk 5 above)
Comment (JulianWielga) [reply ui/client/cypress.json:16]: same as above ;)

File: ui/client/cypress.json (hunk 3 above)
Comment (JulianWielga) [reply ui/client/cypress.json:11]: same as below :)

Comment (lciolecki) [reply ui/client/cypress.json:11]: Ahh! You're right :)

File: ui/client/cypress.json (hunk 5 above)
Comment (lciolecki) [reply ui/client/cypress.json:16]: Ahh! You're right :)

File: ui/client/cypress.json (hunk 4 above)
Comment (lciolecki) [reply ui/client/cypress.json:14]: Ahh! You're right :)

File: ui/client/components/modals/CalculateCountsDialog.tsx [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+type RangeButtonProps = {range: Range, onChange: (value: [Moment, Moment]) => void}
+const RangeButton = ({range, onChange}: RangeButtonProps): JSX.Element => {
+  const {from, to, name} = range
+  const onClick = useCallback(
+    () => onChange([from(), to()]),
+    [from, to],
+  )
+  return (
+    <ButtonWithFocus type="button" title={name} className="predefinedRangeButton" onClick={onClick}>
```
Comment (lciolecki) [inline ui/client/components/modals/CalculateCountsDialog.tsx:77]: Can we also refactor `predefinedRangeButton` to `predefined-range-button`? After that we will have more styles in one standard.

File: ui/client/components/modals/GenericModalDialog.tsx [hunk 7]
```diff
@@ -71,12 +71,14 @@ class GenericModalDialog extends React.Component<Props, State> {
       >
         <div className="draggable-container">
           <Draggable bounds="parent" handle=".modal-draggable-handle">
-            <div className={`espModal ${this.props.style || "confirmationModal"}`}>
-              {this.props.header ? (
-                <div className="modal-title modal-draggable-handle" style={{color: "white", backgroundColor: "#70C6CE"}}>
-                  <span>{this.props.header}</span>
-                </div>
-              ) : null}
+            <div className={`espModal ${this.props.style || "confirmationModal"}`} data-testid="modal">
```
Comment (lciolecki) [inline ui/client/components/modals/GenericModalDialog.tsx:74]: Can we also refactor `espModal` to `nussknacker-modal` and the same with `confirmation-modal` to `confirmation-modal`? :)

File: ui/client/components/modals/CalculateCountsDialog.tsx (hunk 6 above)
Comment (JulianWielga) [reply ui/client/components/modals/CalculateCountsDialog.tsx:81]: I will not touch the styles topic now. next pr will bring visual changes

File: ui/client/components/modals/GenericModalDialog.tsx (hunk 7 above)
Comment (JulianWielga) [reply ui/client/components/modals/GenericModalDialog.tsx:74]: I will not touch the styles topic now. next pr will bring visual changes

**Review by lciolecki — APPROVED** [review]
LGTM

**Pull request merged**